faq-site-client/src/utilities/imgur.ts

```diff
@@ -0,0 +1,58 @@
+import axios, {AxiosResponse} from "axios";
+
+const imgurBaseUrl = "https://api.imgur.com/3";
+const imgurClientId = "5b157adc646460e"; // This client ID is linked to CSHub change if not using for that project
+
+// Documentation: https://apidocs.imgur.com
+export class ImgurUpload {
+
+    public static async findAndReplaceImagesWithImgurLinks(delta: object[]): Promise<object> {
+        for (const i in delta) {
+            // Found an image!
+            if (typeof (delta[i] as any).insert.image !== "undefined") {
+                // The base64 of the image
+                const originalBase64 = (delta[i] as any).insert.image;
+                const strippedBase64 = this.stripBase64Preamble(originalBase64); // Strip the preamble (imgur wants this)
+
+                // Upload it to imgur
+                await this.uploadBase64ToImgur(strippedBase64).then((response: AxiosResponse) => {
+                    if (response.data.success) {
+                        // Replace the base64 with the imgur link
+                        (delta[i] as any).insert.image = response.data.data.link;
+                    } else {
+                        // TODO: Error handling, Maybe retry otherwise fail silently(?)
```

Comment:
How should we handle the errors

Reply:
I guess we should retry once, and if that doesn't work show a dialog. 

For that we should probably make a dialog component so we can programmatically add and remove dialogs (not creating a dialog for every piece of text we want to show in a dialog) and handle click events etc. So maybe a yes no dialog and a notification dialog, the other dialogs will be custom (as they probably won't share that much functionality).

Also, you should probably also check for an undefined / null response.